.projenrc.ts

```diff
@@ -374,11 +375,24 @@ new JsiiBuild(cloudAssemblySchema, {
 (() => {
   cloudAssemblySchema.preCompileTask.exec('tsx projenrc/update.ts');
 
+  // This file will be generated at release time. It needs to be gitignored or it will
+  // fail projen's "no tamper" check, which means it must also be generated every build time.
+  cloudAssemblySchema.preCompileTask.exec('[[ -f cli-version.json ]] || echo \'{ "version": "" }\' > cli-version.json');
+  cloudAssemblySchema.gitignore.addPatterns('cli-version.json');
+
   cloudAssemblySchema.addPackageIgnore('*.ts');
   cloudAssemblySchema.addPackageIgnore('!*.d.ts');
   cloudAssemblySchema.addPackageIgnore('** /scripts');
 })();
 
+new InsertTaskStep(repoProject, {
+  taskName: 'release',
+  insertSteps: [
+    { exec: 'ts-node projenrc/copy-cli-version-to-assembly.task.ts' },
+  ],
+  beforeExec: 'yarn workspaces run build',
```

Comment:
I suggest making the necessary changes to `InsertTaskStep` so that invoking it eventually looks like this:

```suggestion
  afterExec: 'yarn workspaces run bump',
```

Since the `bump` tasks is actually what is important here.

Reply:
> I had that first but I find a "before" much more natural for an "insert" operation.

hh so call it `AppendTaskStep` :)

Its fine. just a personal preference.